contracts/src/token/erc721/extensions/consecutive.rs

```diff
@@ -0,0 +1,1054 @@
+        /// Initialization marker. If true this means that consecutive mint was already triggered.
+        bool _initialized
+    }
+}
+
+sol! {
+    /// Emitted when the tokens from `fromTokenId` to `toTokenId` are transferred from `fromAddress` to `toAddress`.
+    ///
+    /// * `fromTokenId` - First token being transferred.
+    /// * `toTokenId` - Last token being transferred.
+    /// * `fromAddress` - Address from which tokens will be transferred.
+    /// * `toAddress` - Address where the tokens will be transferred to.
+    event ConsecutiveTransfer(
+        uint256 indexed fromTokenId,
+        uint256 toTokenId,
+        address indexed fromAddress,
+        address indexed toAddress
+    );
+}
+
+sol! {
+    /// Batch mint is restricted to the constructor.
+    /// Any batch mint not emitting the [`IERC721::Transfer`] event outside of the constructor
+    /// is non ERC-721 compliant.
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+    error ERC721ForbiddenBatchMint();
+
+    /// Exceeds the max number of mints per batch.
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+    error ERC721ExceededMaxBatchMint(uint256 batchSize, uint256 maxBatch);
+
+    /// Individual minting is not allowed.
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+    error ERC721ForbiddenMint();
+
+    /// Batch burn is not supported.
+    #[derive(Debug)]
+    #[allow(missing_docs)]
+    error ERC721ForbiddenBatchBurn();
+}
+
+#[derive(SolidityError, Debug)]
+pub enum Error {
+    /// Error type from erc721 contract [`erc721::Error`]
+    Erc721(erc721::Error),
+    /// Error type from checkpoint contract [`checkpoints::Error`]
+    Checkpoints(checkpoints::Error),
+    /// Batch mint is restricted to the constructor.
+    /// Any batch mint not emitting the [`IERC721::Transfer`] event outside of
+    /// the constructor is non ERC-721 compliant.
+    ForbiddenBatchMint(ERC721ForbiddenBatchMint),
+    /// Exceeds the max amount of mints per batch.
+    ExceededMaxBatchMint(ERC721ExceededMaxBatchMint),
+    /// Individual minting is not allowed.
+    ForbiddenMint(ERC721ForbiddenMint),
+    /// Batch burn is not supported.
+    ForbiddenBatchBurn(ERC721ForbiddenBatchBurn),
+}
+
+impl MethodError for erc721::Error {
+    fn encode(self) -> alloc::vec::Vec<u8> {
+        self.into()
+    }
+}
+
+impl MethodError for checkpoints::Error {
+    fn encode(self) -> alloc::vec::Vec<u8> {
+        self.into()
+    }
+}
+
+// TODO: add an option to override these constants
+
+/// Maximum size of a batch of consecutive tokens. This is designed to limit
+/// stress on off-chain indexing services that have to record one entry per
+/// token, and have protections against "unreasonably large" batches of tokens.
+pub const MAX_BATCH_SIZE: U96 = uint!(5000_U96);
+
+/// Used to offset the first token id in
+/// [`Erc721Consecutive::_next_consecutive_id`]
+pub const FIRST_CONSECUTIVE_ID: U96 = uint!(0_U96);
+
+/// Consecutive extension related implementation:
```

Comment:
Actually would make these more prominent. Something like:

```rust
// ************** Consecutive **************
```

I think we can explore what it would look like.